Source/WebCore/platform/mediastream/RealtimeMediaSourceCapabilities.h

```diff
@@ -49,98 +51,63 @@ class CapabilityValueOrRange {
     };
     Type type() const { return m_type; }
 
-    union ValueUnion {
-        double asDouble;
-        int asInt;
-    };
+    using ValueUnion = std::variant<double, int>;
 
     CapabilityValueOrRange()
-        : m_type(Undefined)
+        : m_type(Type::Undefined)
     {
     }
 
     CapabilityValueOrRange(double value)
-        : m_type(Double)
+        : m_type(Type::Double)
     {
-        m_minOrValue.asDouble = value;
+        m_minOrValue = value;
     }
 
     CapabilityValueOrRange(int value)
-        : m_type(ULong)
+        : m_type(Type::ULong)
     {
-        m_minOrValue.asInt = value;
+        m_minOrValue = value;
     }
 
     CapabilityValueOrRange(double min, double max)
-        : m_type(DoubleRange)
+        : m_type(Type::DoubleRange)
     {
-        m_minOrValue.asDouble = min;
-        m_max.asDouble = max;
+        m_minOrValue = min;
+        m_max = max;
     }
     
     CapabilityValueOrRange(int min, int max)
-        : m_type(ULongRange)
+        : m_type(Type::ULongRange)
     {
-        m_minOrValue.asInt = min;
-        m_max.asInt = max;
+        m_minOrValue = min;
+        m_max = max;
     }
 
     const ValueUnion& rangeMin() const
     {
-        ASSERT(m_type == DoubleRange || m_type == ULongRange);
+        ASSERT(m_type == Type::DoubleRange || m_type == Type::ULongRange);
         return m_minOrValue;
     }
 
     const ValueUnion& rangeMax() const
     {
-        ASSERT(m_type == DoubleRange || m_type == ULongRange);
+        ASSERT(m_type == Type::DoubleRange || m_type == Type::ULongRange);
         return m_max;
     }
 
     const ValueUnion& value() const
     {
-        ASSERT(m_type == Double || m_type == ULong);
+        ASSERT(m_type == Type::Double || m_type == Type::ULong);
         return m_minOrValue;
     }
 
-    template<class Encoder> void encode(Encoder&) const;
-    template<class Decoder> static WARN_UNUSED_RETURN bool decode(Decoder&, CapabilityValueOrRange&);
-
 private:
     ValueUnion m_minOrValue { };
     ValueUnion m_max { };
-    Type m_type { Undefined };
+    Type m_type { Type::Undefined };
```

Comment:
I think we should replace all of this with a std::variant<int, double, std::pair<int, int>, std::pair<double, double>> with maybe a std::nullptr_t if Type::Undefined is still needed.  Otherwise we can still send a type and values that don't line up.